include/knowhere/bitsetview.h

```diff
@@ -50,7 +50,8 @@ class BitsetView {
 
     bool
     test(int64_t index) const {
-        return bits_[index >> 3] & (0x1 << (index & 0x7));
+        // when index is larger than the max_offset, ignore it
+        return (bits_[index >> 3] & (0x1 << (index & 0x7))) || (index >= static_cast<int64_t>(num_bits_));
```

Comment:
Can I have the code point of the validation?

Some questions:
1. If index is out of range, `bits_[index >> 3]` will overflow and the result before `||` is undefined, making the range check get ignore. Is this as expected?
2. Will the `active_count` get updated in the chunk granularity?

Reply:
1. Good catch !  I set the range check at the right side because it is just a corner case, and once the left checking success, then the right side checking can be ignored, which can reduce the influence of cost at most . But it seems that the range checking is better placed at the left side to avoid overflow.
2. `active_count` is just the count of visible vectors in the segments, which can be determined just by the [`Timestamp`.](https://github.com/milvus-io/milvus/blob/45d9fb5929e400f42f7d35d51046189089b2ad85/internal/core/src/query/visitors/ExecPlanNodeVisitor.cpp#L91) . This value is a determined by the search context. For other cases such as sealed index, it is a const property and never be changed.